src/wallet/test/wallet_tests.cpp

```diff
@@ -301,10 +312,16 @@ class ListCoinsTestingSetup : public TestChain100Setup
             blocktx = CMutableTransaction(*wallet->mapWallet.at(tx->GetHash()).tx);
         }
         CreateAndProcessBlock({CMutableTransaction(blocktx)}, GetScriptForRawPubKey(coinbaseKey.GetPubKey()));
-        LOCK(wallet->cs_wallet);
-        auto it = wallet->mapWallet.find(tx->GetHash());
-        BOOST_CHECK(it != wallet->mapWallet.end());
-        it->second.SetMerkleBranch(chainActive.Tip(), 1);
+        std::map<uint256, CWalletTx>::iterator it;
+        {
```

Comment:
Can you just take both locks instead of adding a new scope and lots of diff lines? Its in a test, locks shouldn't matter much.